### PR TITLE
Prefer const

### PR DESCRIPTION
**Pull request by stbaer**
- added eslint rule prefer-const
- used eslint --fix to fix most warnings, changed the others manually

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
Nice :+1:

File: src/core/math/Matrix.js [hunk 1]
```diff
@@ -10,10 +10,8 @@ import Point from './Point';
  * @class
  * @memberof PIXI
  */
-class Matrix
-{
-    constructor()
-    {
+class Matrix {
+    constructor() {
```
Comment (bigtimebuddy) [inline src/core/math/Matrix.js]: Style comment, curly braces should be on a new line.

File: src/core/math/Matrix.js [hunk 2]
```diff
@@ -53,392 +51,369 @@ class Matrix
...
+     * a = array[0]
+     * b = array[1]
+     * c = array[3]
+     * d = array[4]
+     * tx = array[2]
+     * ty = array[5]
+     *
+     * @param array {number[]} The array that the matrix will be populated from.
+     */
+    fromArray(array) {
```
Comment (bigtimebuddy) [inline src/core/math/Matrix.js]: Curly braces on a new line.

File: src/core/math/Matrix.js [hunk 3]
```diff
@@ -53,392 +51,369 @@ class Matrix
...
+     * @param {number} b
+     * @param {number} c
+     * @param {number} d
+     * @param {number} tx
+     * @param {number} ty
+     *
+     * @return {PIXI.Matrix} This matrix. Good for chaining method calls.
+     */
+    set(a, b, c, d, tx, ty) {
+        this.a = a;
```
Comment (bigtimebuddy) [inline src/core/math/Matrix.js]: Curly braces on a new line.

File: src/core/math/Matrix.js [hunk 4]
```diff
@@ -53,392 +51,369 @@ class Matrix
...
+
+
+    /**
+     * Creates an array from the current Matrix object.
+     *
+     * @param transpose {boolean} Whether we need to transpose the matrix or not
+     * @param [out=new Float32Array(9)] {Float32Array} If provided the array will be assigned to out
+     * @return {number[]} the newly created array which contains the matrix
+     */
+    toArray(transpose, out) {
```
Comment (bigtimebuddy) [inline src/core/math/Matrix.js]: Curly braces on a new line. Basically there are a lot of these. I won't comment on all of them.

File: src/core/renderers/SystemRenderer.js [hunk 5]
```diff
@@ -38,7 +38,7 @@ class SystemRenderer extends EventEmitter
         // prepare options
         if (options)
         {
-            for (let i in CONST.DEFAULT_RENDER_OPTIONS)
+            for (const i in CONST.DEFAULT_RENDER_OPTIONS)
```
Comment (bigtimebuddy) [inline src/core/renderers/SystemRenderer.js]: How is this valid? Doesn't `i` get re-assigned here every time?

**Comment by stbaer** [comment]
Oh ya, will check if theres also a eslint rule for the curly braces.

Think the for ... const comes from eslint autofix. At least I hope I didn't do that :) will check later, only got my phone at the Moment.

**Comment by stbaer** [comment]
Btw, what do you think about a lint-fix npm script?

File: src/core/renderers/SystemRenderer.js (hunk 5 above)
Comment (englercj) [inline src/core/renderers/SystemRenderer.js]: the iterator can be const in `for-in` statements and `for-of` statements, just not in `for(;;)` statements.

> ForDeclaration : LetOrConst ForBinding

http://www.ecma-international.org/ecma-262/6.0/index.html#sec-for-in-and-for-of-statements-static-semantics-boundnames

**Comment by englercj** [comment]
> Oh ya, will check if theres also a eslint rule for the curly braces.

There is, it was in my recommended pixi starting eslint file:

https://gist.github.com/englercj/27edc75f0aa2b004f2bc6736454d3411#file-eslintrc-json-L147

**Comment by stbaer** [comment]
+1 for the eslintrc @englercj (no github reactions in the phn...) but without the comments maybe bc it's not valid json.

**Comment by stbaer** [comment]
Might also be worth thinking about a seperated eslint package that all plugins should include also.

**Comment by englercj** [comment]
> but without the comments maybe bc it's not valid json.

eslintrc.json files allow comments, its fine I promise :)

> Might also be worth thinking about a seperated eslint package that all plugins should include also.

We don't want to enforce plugin authors to use any style. 1st-party plugins will be in our repo, and 3rd-party plugins can look however their authors want.

**Comment by stbaer** [comment]
:+1:

**Comment by englercj** [comment]
Initiating operation merge conflicts. GO GO GO

**Pull request merged**